src/transaction/deferred_action_thread.cpp

```diff
@@ -0,0 +1,11 @@
+#include "transaction/deferred_action_thread.h"
+#include "transaction/deferred_action_manager.h"
+
+namespace terrier::transaction {
+    void DeferredActionThread::DeferredActionThreadLoop() {
+    while (run_deferred_events_) {
+      std::this_thread::sleep_for(deferred_actions_period_);
```

Comment:
I don't think we want to unconditionally sleep.  @GustavoAngulo can probably provide insight on how to efficiently check whether sleeping is appropriate since he had to address that issue in the log manager.